packages/core/src/verify.ts

```diff
@@ -0,0 +1,77 @@
+import { IgnitionError } from "./errors";
+import { ERRORS } from "./errors-list";
+import { builtinChains } from "./internal/chain-config";
+import { FileDeploymentLoader } from "./internal/deployment-loader/file-deployment-loader";
+import { encodeDeploymentArguments } from "./internal/execution/abi";
+import { loadDeploymentState } from "./internal/execution/deployment-state-helpers";
+import { assertIgnitionInvariant } from "./internal/utils/assertions";
+import { findDeployedContracts } from "./internal/views/find-deployed-contracts";
+import { VerifyResult } from "./types/verify";
+
+/**
+ * Retrieve the information required to verify all contracts from a deployment on Etherscan.
+ *
+ * @param deploymentDir - the file directory of the deployment
+ *
+ * @beta
+ */
+export async function verify(deploymentDir: string): Promise<VerifyResult> {
+  const deploymentLoader = new FileDeploymentLoader(deploymentDir);
+
+  const deploymentState = await loadDeploymentState(deploymentLoader);
+
+  if (deploymentState === undefined) {
+    throw new IgnitionError(ERRORS.VERIFY.UNINITIALIZED_DEPLOYMENT, {
+      deploymentDir,
+    });
+  }
+
+  const contracts = Object.entries(findDeployedContracts(deploymentState));
+
+  if (contracts.length === 0) {
+    throw new IgnitionError(ERRORS.VERIFY.NO_CONTRACTS_DEPLOYED, {
+      deploymentDir,
+    });
+  }
+
+  const chainConfig = builtinChains.find(
+    (c) => c.chainId === deploymentState.chainId
+  );
+
+  // this case is a failure on our part
```

Comment:
The main difference with hardhat-verify is that it allows you to set a new chain in your config. Maybe `verify` in core can take an optional chain config, and the hardhat plugin allow you to set it in the hardhat config, and then pass it to core.

I think this is important because it's common to deploy the same thing to multiple chains.

WDYT?

Reply:
I think we should expand the list of chains to match the HH-viem list, and support the "other networks" config setup that it supports:

https://hardhat.org/hardhat-runner/plugins/nomicfoundation-hardhat-verify#adding-support-for-other-networks